flipperdevices/flipperzero-firmware
# nfc: NTAG password auto capture (and other password-related changes)

# What's new

- Add flow to automatically capture NTAG PWD from reader
- Miscellaneous MFUL bug fixes
- Default MFUL manual password entry value is 0xFFFFFFFF (as shipped on tags)
  - Note: a simple read of the tag never captures the PWD and PACK because they are blanked out by the tag during read, so the unlock options will always be available for a newly read tag.
- Show PWD and PACK on tag info page if available
  - Useful if you need to write to an emulated tag with something and need the password to unlock it before writing, since password authentication is faithfully emulated
- Add options to unlock MFUL from loaded file
  - You could certainly just read the card from scratch again, and currently it still asks you to save to a different file, so some discussion might be needed whether this is useful or whether it should auto save back to the original file after unlock

Supercedes #1471

The bug in #1842 is present, this PR **does not** attempt to address the issue

# Verification 

- Automatic NTAG PWD capture flow
  1. Read password-protected NTAG as usual
  2. Select "Unlock", then "Unlock With Reader"
  3. Flipper emulates read card so far; tap Flipper to original reader the card was supposed to be used with
  4. If the reader sent a PWD_AUTH command, Flipper will beep and continue after it has been removed from reader
  5. Flow resumes to "Unlock With Password" flow after entering password; following through should result in the matching card being completely read
 - Default MFUL manual password entry value is 0xFFFFFFFF
   1. Read password-protected NTAG as usual
   2. Select "Unlock With Password"
   3. Select "Enter Password Manually"
   4. The password should show `FF FF FF FF` by default
- Show PWD and PACK on tag info page if available
  1. Load an NTAG NFC file that has been unlocked with password
  2. Select "Info"
  3. Scroll down; Password and PACK should be displayed
- Add options to unlock MFUL from loaded file
  1. Load an NTAG NFC file that is password protected but has not been unlocked
  2. "Unlock" option should appear in the menu and function as if the card was newly read

# Checklist (For Reviewer)
